Makefile

```diff
@@ -143,34 +140,43 @@ endif
 
 DOCKER_RUN_DOCKER := $(DOCKER_FLAGS) "$(DOCKER_IMAGE)"
 
+DOCKER_BUILD_ARGS += --build-arg=GO_VERSION
+BUILD_OPTS := ${BUILD_APT_MIRROR} ${DOCKER_BUILD_ARGS} ${DOCKER_BUILD_OPTS} -f "$(DOCKERFILE)"
+ifdef USE_BUILDX
+BUILD_OPTS += $(BUILDX_BUILD_EXTRA_OPTS)
+BUILD_CMD := $(BUILDX) build
+else
+BUILD_CMD := $(DOCKER) build
+endif
+
+ifdef DOCKER_CROSSPLATFORMS
+DOCKER_CROSS = true
+endif
+
 default: binary
 
 all: build ## validate all checks, build linux binaries, run all tests\ncross build non-linux binaries and generate archives
 	$(DOCKER_RUN_DOCKER) bash -c 'hack/validate/default && hack/make.sh'
 
-binary: DOCKER_BUILD_ARGS += --output=bundles/ --target=binary
-binary: build ## build the linux binaries
+binary: buildx ## build the linux binaries
+	$(BUILD_CMD) $(BUILD_OPTS) --output=bundles/ --target=binary .
 
-dynbinary: DOCKER_BUILD_ARGS += --output=bundles/ --target=dynbinary
-dynbinary: build ## build the linux dynbinaries
+dynbinary: buildx ## build the linux dynbinaries
+	$(BUILD_CMD) $(BUILD_OPTS) --output=bundles/ --target=dynbinary .
 
-cross: DOCKER_BUILD_ARGS += --output=bundles/ --target=cross --build-arg DOCKER_CROSSPLATFORMS=$(DOCKER_CROSSPLATFORMS)
-cross: DOCKER_CROSS := true
-cross: build ## cross build the binaries for darwin, freebsd and\nwindows
+cross: buildx ## cross build the binaries for darwin, freebsd and\nwindows
+	$(BUILD_CMD) $(BUILD_OPTS) --output=bundles/ --target=cross --build-arg DOCKER_CROSSPLATFORMS --build-arg CROSS=true .
```

Comment:
these three can probably be reduced to something like this:
```make
binary dynbinary: ## build the linux binaries
cross: ## cross build the binaries for darwin, freebsd and\nwindows

cross: BUILD_OPTS += --build-arg DOCKER_CROSSPLATFORMS --build-arg CROSS=true

binary dynbinary cross: buildx
	$(BUILD_CMD) $(BUILD_OPTS) --output=bundles/ --target=$@
```

(and I hope won't reintroduce the same problem we're trying to fix here)

Reply:
It would re-introduce the issue (I think).
I really don't like updating global variables with target specific data.

Reply:
> It would re-introduce the issue (I think)

since we no longer have an intermediary dep, I think it won't...

test (with patch)
```
make BIND_DIR=. binary dynbinary cross
...
docker build  --build-arg=GO_VERSION  -f "Dockerfile" --output=bundles/ --target=binary .
...
docker build  --build-arg=GO_VERSION  -f "Dockerfile" --output=bundles/ --target=dynbinary .
...
docker build  --build-arg=GO_VERSION  -f "Dockerfile" --build-arg DOCKER_CROSSPLATFORMS --build-arg CROSS=true --output=bundles/ --target=cross .
```

Reply:
Nice (and surprised), updated.